src/registry_api.rs

```diff
@@ -227,4 +247,71 @@ impl RegistryApi {
 
         Ok(result)
     }
+
+    /// Fetch crates from the registry's API
+    pub(crate) async fn get_crates(&self, query: Option<&str>) -> Result<Search> {
+        #[derive(Deserialize, Debug)]
+        struct SearchError {
+            detail: String,
+        }
+
+        #[derive(Deserialize, Debug)]
+        struct SearchResponse {
+            crates: Option<Vec<SearchCrate>>,
+            meta: Option<SearchMeta>,
+            errors: Option<Vec<SearchError>>,
+        }
+
+        let url = {
+            let mut url = self.api_base.clone();
+            url.path_segments_mut()
+                .map_err(|()| anyhow!("Invalid API url"))?
+                .extend(&["api", "v1", "crates"]);
+            url.set_query(query);
+            url
+        };
+
+        // Extract the query from the query args
+        let executed_query = url.query_pairs().find_map(|(key, value)| {
```

Comment:
I wonder if we could remove this. 

From what I see in the search handler, `get_crates` is always called with already expanded `query` we get here, so we can just directly get it in the handler.

Reply:
Possible.

Turned out there was already an extraction step on the `paginate` parameters in the search handler so I only had to modify that to extract `sort` and `q` instead of just `sort`.